src/Control/Email/Email.php

```diff
@@ -753,215 +370,145 @@ public function getHTMLTemplate()
-        if (!$htmlPart && $htmlTemplate && (!$plainOnly || empty($plainPart))) {
-            $htmlPart = $this->renderWith($htmlTemplate, $this->getData());
+        if ($htmlBody && !$dataChangedSincedLastRender) {
+            $htmlRender = $htmlBody;
         }
 
-        // Plain part fails over to generated from html
-        if (!$plainPart && $htmlPart) {
-            /** @var DBHTMLText $htmlPartObject */
-            $htmlPartObject = DBField::create_field('HTMLFragment', $htmlPart);
-            $plainPart = $htmlPartObject->Plain();
+        if ($plainBody && !$dataChangedSincedLastRender) {
+            $plainRender = $plainBody;
         }
 
-        // Rendering is finished
-        Requirements::restore();
+        // Render plain
+        if (!$plainRender && $plainTemplate) {
+            $plainRender = ViewableData::create()->renderWith($plainTemplate, $this->getData())->Plain();
+        }
 
-        // Fail if no email to send
-        if (!$plainPart && !$htmlPart) {
-            return $this;
+        // Render HTML, either if sending html email, or a plain part is lacking
+        if (!$htmlRender && $htmlTemplate && (!$plainOnly || empty($plainRender))) {
+            $htmlRender = ViewableData::create()->renderWith($htmlTemplate, $this->getData())->RAW();
         }
 
-        // Build HTML / Plain components
-        if ($htmlPart && !$plainOnly) {
-            $this->setBody($htmlPart);
-            $this->getSwiftMessage()->setContentType('text/html');
-            $this->getSwiftMessage()->setCharset('utf-8');
-            if ($plainPart) {
-                $this->getSwiftMessage()->addPart($plainPart, 'text/plain', 'utf-8');
-            }
-        } else {
-            if ($plainPart) {
-                $this->setBody($plainPart);
-            }
-            $this->getSwiftMessage()->setContentType('text/plain');
-            $this->getSwiftMessage()->setCharset('utf-8');
+        // Plain render fails over to generated from html
+        if (!$plainRender && $htmlRender) {
+            // call html_entity_decode() to ensure any encoded HTML is also stripped inside ->Plain()
+            $dbField = DBField::create_field('HTMLFragment', html_entity_decode($htmlRender));
+            $plainRender = $dbField->Plain();
         }
 
-        return $this;
-    }
+        // Remove the $htmlRender if was only used as fail over to create $plainRender
+        if ($plainOnly && $htmlRender) {
+            $htmlRender = null;
+        }
 
-    /**
-     * @return Swift_MimePart|false
-     */
-    public function findPlainPart()
-    {
-        foreach ($this->getSwiftMessage()->getChildren() as $child) {
-            if ($child instanceof Swift_MimePart && $child->getContentType() == 'text/plain') {
-                return $child;
-            }
+        // Handle edge case where data changed since last render and there's no template
+        if (!$htmlRender && $htmlBody) {
+            $htmlRender = $htmlBody;
```

Comment:
Should this be done if `$plainOnly` is true?

Reply:
It now gets reverted later on in `$plainOnly` is true